datafusion/sqllogictest/test_files/array.slt

```diff
@@ -5260,6 +5270,13 @@ select array_has([], null),
 ----
 NULL NULL NULL
 
+# If lhs is has any Nulls, we return Null instead of false
+query BB
+select array_has([1, null, 2], 3),
```

Comment:
I'm not sure this is a correct behavior.

Checking DuckDB

```
D select array_has([1, null, 2], 3);
┌───────────────────────────────────────────┐
│ array_has(main.list_value(1, NULL, 2), 3) │
│                  boolean                  │
├───────────────────────────────────────────┤
│ false                                     │
└───────────────────────────────────────────┘
```

Reply:
Hmm DuckDB is different then Spark and Postgres then

```
spark-sql (default)> SELECT array_contains(array(1, NULL, 3), 4);
NULL
```

```
postgres=# SELECT 4 =ANY('{1, NULL, 3}');
 ?column? 
----------
 
(1 row)
```

Was hoping to use this to implement `ArrayContains` for Comet, and this is the one behavior difference.

Reply:
Turns out the existing "correct" behavior for scalars doesn't match DuckDB either, which never returns null unless the whole array is null

DuckDB
```
D select array_has([null, null], 4);
┌───────────────────────────────────────────┐
│ array_has(main.list_value(NULL, NULL), 4) │
│                  boolean                  │
├───────────────────────────────────────────┤
│ false                                     │
└───────────────────────────────────────────┘
```

DF
```
DataFusion CLI v43.0.0
> select array_has([null, null], 4);
+-------------------------------------------+
| array_has(make_array(NULL,NULL),Int64(4)) |
+-------------------------------------------+
|                                           |
+-------------------------------------------+
1 row(s) fetched. 
Elapsed 0.001 seconds.
```

Reply:
Updated to match DuckDB behavior and updated description